src/JsonSchema/Constraints/CollectionConstraint.php

```diff
@@ -63,23 +63,29 @@ protected function validateItems($value, $schema = null, JsonPointer $path = nul
     {
         if (is_object($schema->items)) {
             // just one type definition for the whole array
-            foreach ($value as $k => $v) {
-                $initErrors = $this->getErrors();
+            if (SimpleItemsConstraint::isSupported($schema)) {
```

Comment:
I find this a little inconsistent with the rest of the codebase... the constraint classes don't generally have functional dependencies on each other outside of methods specified in `ConstraintInterface` ( eg. `check()` ). Is a new constraint type really necessary? What do you think about just doing everything you need to do inside of `CollectionConstraint`?